projects/fw_103_hw/src/main.c

```diff
@@ -0,0 +1,69 @@
+#include <stdbool.h>
+#include <stdio.h>
+
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+
+// GPIO address that we will allow us to read button data
+const GpioAddress potentiometer_addr_A6 = { // could be passed as context
```

Comment:
Good programming practice is to not to use global variables (ie any variable that is non-static declared outside a function). The const is good since they won't be changed, but declare them static as well `const static`